FluentEventWriter.cpp

```diff
@@ -0,0 +1,117 @@
+/*
+    microsoft-oms-auditd-plugin
+
+    Copyright (c) Microsoft Corporation
+
+    All rights reserved.
+
+    MIT License
+
+    Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the ""Software""), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+
+    The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+
+    THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+*/
+#include "FluentEventWriter.h"
+
+void FluentEventWriter::write_int32_field(const std::string& name, int32_t value)
+{
+    _recordFields[name] = std::to_string(value);
+}
+
+void FluentEventWriter::write_int64_field(const std::string& name, int64_t value)
+{
+    _recordFields[name] = std::to_string(value);
+}
+
+void FluentEventWriter::write_raw_field(const std::string& name, const char* value_data, size_t value_size)
+{
+    _recordFields[name] = std::string(value_data, value_size);
+}
+
+
+bool FluentEventWriter::begin_event(const Event& event)
+{
+    _fluentEvent = new FluentEvent(_tag);
+    _eventCommonFields.clear();
+
+    char hostname[HOST_NAME_MAX];
+    gethostname(hostname, HOST_NAME_MAX);
```

Comment:
Add a hostname value to TextEventWriterConfig and then use that value.
Add a test value in TestEventData.cpp
Set the hostname value in FluentEentWriterTests.cpp from the value in TestEventData.cpp
Then, fix the fluent_test_events in TestEventData.cpp so that Computer has the value set in the unit test.

Reply:
Fixed the Computer value issue in FluentEentWriterTests.cpp

---
In reply to: [399427008](https://github.com/microsoft/OMS-Auditd-Plugin/pull/43#discussion_r399427008) [](ancestors = 399427008)

Reply:
Incorporated the suggested changes

---
In reply to: [399526710](https://github.com/microsoft/OMS-Auditd-Plugin/pull/43#discussion_r399526710) [](ancestors = 399526710,399427008)